### PR TITLE
Developer guide for code style consistency 

### PR DESCRIPTION
**Pull request by r2avula**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Docs update and minor refactoring

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

* **Other information**:

### CODE REVIEW

File: waywiser_carla/README.md
Comment (ariamirzai) [inline waywiser_carla/README.md:1]: Add reference in WayWIseR main guide? E.g. "if interested in Carla simulation, follow instructions for installation here ..."

File: README.md
Comment (ariamirzai) [inline README.md:1]: Should just mention that gazebo is optional and refer to the readme in waywiser_gazebo instead? Becomes confusing to have Gazebo instructions in two different places.

File: waywiser_carla/README.md
Comment (r2avula) [reply waywiser_carla/README.md:1]: I already have this in main readme file: https://github.com/RISE-Dependable-Transport-Systems/WayWiseR/blob/efa01208c9440453e289091434e488e7d43104c7/README.md?plain=1#L69
Is this what you mean?

File: README.md
Comment (r2avula) [reply README.md:1]: I agree. I intend to have instructions only at one place (in the readme files in the packages dedicated to gazebo and carla). I removed their related text from the [main readme file ](https://github.com/RISE-Dependable-Transport-Systems/WayWiseR/blob/feature/dev-guide/README.md#how-to-install-and-build-on-ubuntu-2204) and moved them to [waywiser_gazebo](https://github.com/RISE-Dependable-Transport-Systems/WayWiseR/blob/feature/dev-guide/waywiser_gazebo/README.md#gazebo-setup) and [waywiser_carla](https://github.com/RISE-Dependable-Transport-Systems/WayWiseR/blob/feature/dev-guide/waywiser_carla/README.md#carla-setup). 

Are there still any left over instructions in the [main readme file](https://github.com/RISE-Dependable-Transport-Systems/WayWiseR/blob/feature/dev-guide/README.md) related to gazebo?

File: .github/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Developer Guide
+
+This guide sets up a development environment with [Visual Studio Code](https://code.visualstudio.com/) as the recommended IDE. This setup ensures code style consistency and easy testing within VS Code.
+
+## Prerequisites
+
+Install these VS Code extensions:
+
+- [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+- [Ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)
```
Comment (ariamirzai) [inline .github/DEVELOPER_GUIDE.md:10]: "isort" extension is only a preview version and could contain issues, just so you know.xsl

File: .github/DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+[format]
+quote-style = "single"
+```
+
+## Configure Uncrustify for C++
+
+
+1. Download [ament_code_style.cfg](https://github.com/ament/ament_lint/blob/humble/ament_uncrustify/ament_uncrustify/configuration/ament_code_style.cfg).
+2. Save it in `.vscode/ament_code_style.cfg`.
```
Comment (ariamirzai) [inline .github/DEVELOPER_GUIDE.md]: Replace these descriptions with following command:
- - - - - - - - - -
Download amend_code_style.gfg to the .vscode directory: 

wget https://github.com/ament/ament_lint/blob/humble/ament_uncrustify/ament_uncrustify/configuration/ament_code_style.cfg
- - - - - - - - - -

**Review by ariamirzai — CHANGES_REQUESTED** [review]
Build error (I have cmake version 3.22.1 on agrarsense simulator):
 
CMake Error at CMakeLists.txt:8 (cmake_policy):
  Policy "CMP0148" is not known to this version of CMake.

**Review by ariamirzai — APPROVED** [review]
Tested and works!

**Pull request merged**